onnxruntime/core/providers/cpu/ml/array_feature_extractor.cc

```diff
@@ -53,36 +53,43 @@ template <typename T>
 common::Status ArrayFeatureExtractorOp<T>::Compute(OpKernelContext* context) const {
   const Tensor& X = *context->Input<Tensor>(0);
   const TensorShape& x_shape = X.Shape();
-  const vector<int64_t>& x_dims = x_shape.GetDims();
+  const int64_t x_num_dims = x_shape.NumDimensions();
   const T* x_data = X.template Data<T>();
 
-  if (x_dims.empty()) {
+  if (x_num_dims == 0) {
```

Comment:
So, X cannot be a scalar?

Reply:
i'm not sure. @pranavsharma is that the case?

Reply:
The spec doesn't say explicitly that it can be a scalar. https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.ArrayFeatureExtractor